### PR TITLE
Issue 745 extract iterate state

### PR DESCRIPTION
**Pull request by godber**
Extract iterateState and friends out to stateUtils

My pending k8s backed cluster manager will need to share the
iterateState function and its associated functions so I extract them out
here.

closes https://github.com/terascope/teraslice/issues/745

### CODE REVIEW

File: lib/cluster/services/cluster/backends/stateUtils.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                    clearTimeout(timer);
+                    const formatedData = results.map(formatResponse);
+                    const sortedData = _.sortBy(formatedData, ['name', 'started']);
+                    const reversedData = sortedData.reduce((prev, curr) => {
+                        prev.push(curr);
+                        return prev;
+                    }, []);
+                    resolve(reversedData);
+                })
+                .catch(err => reject({ message: parseError(err), code: 500 }));
```
Comment (peterdemartini) [inline lib/cluster/services/cluster/backends/stateUtils.js]: We should clear the timeout in the catch as well.

File: lib/cluster/services/cluster/backends/stateUtils.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
+'use strict';
```
Comment (peterdemartini) [inline lib/cluster/services/cluster/backends/stateUtils.js]: We should rename this `state-utils.js`

File: spec/lib/cluster/services/cluster/backends/stateUtils-spec.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+            const getSlicerStats = stateUtils.newGetSlicerStats(clusterState, context, messaging);
+
+            try {
+                slicerStats = await getSlicerStats(
+                    ['8d7669e1-d351-4a0d-b36e-b11da5ade26a'],
+                    '8d7669e1-d351-4a0d-b36e-b11da5ade26a'
+                );
+            } catch (e) {
+                console.log(e);
```
Comment (peterdemartini) [inline spec/lib/cluster/services/cluster/backends/stateUtils-spec.js]: We should use expect to make sure this error shouldn't happen. For example, `expect(e).not.toBeDefined()`

**Comment by godber** [comment]
Thanks for the feedback Peter, I've pushed my changes.

**Pull request merged**